### PR TITLE
Try to restore NuGet packages before building projects

### LINKED ISSUE #50: No restore on generation
Generator should restore projects, otherwise it throws errors until projects are restored manually.

### PR DESCRIPTION
**Pull request by Saancreed**
~~Highest quality reflection code :see_no_evil:~~

Also import latest `.editorconfig` rules from CoreLib because analyzers annoyed me far too much while working on this.

Closes #50.

### CODE REVIEW

**Comment by Saancreed** [comment]
Not yet because testing this in any meaningful manner would require us to reference a package in tests that comes from external NuGet feed.

But I suspect it can be done, just not today :slightly_smiling_face:

**Comment by jakubfijalkowski** [comment]
> Not yet because testing this in any meaningful manner would require us to reference a package in tests that comes from external NuGet feed.

Just an idea - delete the `obj` and `bin` folder in the tests. xD

**Review by jakubfijalkowski — APPROVED** [review]

**Comment by Saancreed** [comment]
> Just an idea - delete the `obj` and `bin` folder in the tests. xD

Sure, I can do that. Won't be as definitive as doing `dotnet nuget locals all --clear` but won't hurt us either.

One last thing: currently I also have `RestoreProjectsSerializedAsync` that calls a separate `RestoreAsync` on each input project which I used to debug issues with `DependencyGraphSpec` not being fully merged by the `Union` method (it merges projects but skips restores, forcing us to re–add them manually afterwards). Can we let it stay in case we find some weird behavior while using this feature on merged graphs or should I dump it now (and then inline the `CreateMSBuildStaticGraphRestore` of course)?

**Comment by Saancreed** [comment]
Now with 100% less reflection and +Inf% more support for user–defined BeforeRestore targets!

File: examples/project/embedded/LeanCode.Contracts.pb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+embedded
```
Comment (shilangyu) [inline examples/project/embedded/LeanCode.Contracts.pb:2]: Shouldn't this be in LFS? I'm having a hard time updating the submodule in the dart generator. It complains about git LFS. Not sure if I am doing something wrong or this was committed incorrectly @Saancreed @jakubfijalkowski

Comment (Saancreed) [reply examples/project/embedded/LeanCode.Contracts.pb:2]: I'm indifferent. This particular file is just 42 bytes long, that's even smaller than the `DTO.cs` next to it :sweat_smile:

Well, it does contain few unprintable characters but I wouldn't call this a good reason to migrate it to _large_ file storage.

Comment (jakubfijalkowski) [reply examples/project/embedded/LeanCode.Contracts.pb:2]: It is 42 bytes, so let's just migrate it to LFS as a normal commit.

**Pull request merged**